include/libtorrent/aux_/aligned_union.hpp

```diff
@@ -37,7 +37,7 @@ POSSIBILITY OF SUCH DAMAGE.
 
 namespace libtorrent { namespace aux {
 
-#if defined __GNUC__ && __GNUC__ < 5
+#if defined __GNUC__ && __GNUC__ < 5 && !defined(__clang__)
```

Comment:
technically, this should check for libc++. it's possible to use libstdc++ with clang too

Reply:
true, but I think it's very unlikely that a `libstdc++` corresponding to such and old `gcc` will play nice with `clang` and c++11. What do you think?

Reply:
when clang first appeared on mac, it used libstdc++ by default. and clang has always had strong c++11 support, so I don't think it's that unlikely